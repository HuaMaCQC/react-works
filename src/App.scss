$color-primary-0: #2a94d6; // Main Primary color */
$color-primary-1: #344a5f;
$color-primary-2: #dbe6ec;
$color-primary-3: #f0f1f2;
$color-primary-4: #ffffff;
$color-primary-1-rgba: rgba(42, 148, 214, 1);
$color-primary-2-rgba: rgba(51, 73, 94, 1);
$color-primary-3-rgba: rgba(219, 230, 236, 1);
$color-primary-4-rgba: rgba(239, 240, 241, 1);
$color-primary-5-rgba: rgba(255, 255, 255, 1);

body {
  font-family: Microsoft JhengHei;
}

/* Big Tablet to 1200px (width smaller than 1140px row */
@media only screen and (max-width: 1200px) {}

/* Small Tablet to Big Tablet from 768 to 1023px */
@media only screen and (max-width: 1023px) {}

/* Small Phones to small Tablet 481-767px(Ipad) */
@media only screen and (max-width: 767px) {}

/* Small Phones 0 to 480px(Iphone) */
@media only screen and (max-width: 480px) {}
