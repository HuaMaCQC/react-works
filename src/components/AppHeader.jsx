import React from 'react';
import { Label, Icon } from 'semantic-ui-react';
import styles from './AppHeader.styledJsx.scss';

function resolveScopedStyles(scope) {
  return scope.props.className;
}

export default function AppHeader(props) {
  const { switchMenu, user } = props;
  const jsxId = resolveScopedStyles(
    <scope>
      <style jsx>{styles}</style>
    </scope>,
  );

  return (
    <div className="header">
      <div className="menu-icon">
        <Icon data-testid="switchMenu" link name="sidebar" size="big" onClick={switchMenu} />
      </div>
      <span className="title">管理系統</span>
      <div className="user">
        <div className="notify-bar">
          <Label as="a" size="mini" className={`${jsxId} header-bnt`}>
            <Icon size="large" name="envelope" />
            信件
            <Label circular color="red" className={`${jsxId} user-floating`} floating>
              22
            </Label>
          </Label>
          <Label as="a" size="mini" className={`${jsxId} header-bnt`}>
            <Icon size="large" name="comment" />
            訊息
            <Label circular color="red" className={`${jsxId} user-floating`} floating>
              6
            </Label>
          </Label>
        </div>
        <span className="user-title">歡迎:</span>
        <Label as="a" size="large" className={`${jsxId} header-bnt`}>
          <Icon size="large" name="user circle" />
          {user.name}
        </Label>
      </div>
      <style jsx="true">{styles}</style>
    </div>
  );
}
