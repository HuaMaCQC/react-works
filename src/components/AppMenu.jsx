import React, { useState } from 'react';
import { Accordion, Icon, Menu } from 'semantic-ui-react';
import { NavLink } from 'react-router-dom';
import styles from './AppMenu.styledJsx.scss';

function activeUr(selected, id) {
  const actives = [...selected];
  const idx = actives.indexOf(id);

  if (idx === -1) {
    actives.push(id);
  }

  if (idx !== -1) {
    actives.splice(idx, 1);
  }

  return actives;
}

export default function AppMenu(props) {
  const { menu } = props;
  const [selected, setSelected] = useState([]);

  return (
    <div className="menu">
      <Accordion>
        {
          menu.map(item => (
            <div className="menu-item" key={item.id}>
              <Accordion.Title
                index={0}
                data-id={item.id}
                onClick={() => setSelected(activeUr(selected, item.id))}
                active={selected.indexOf(item.id) !== -1}
                data-testid="title"
              >
                {item.name}
                <div className="icon"><Icon name="dropdown" /></div>
              </Accordion.Title>
              <Accordion.Content data-testid="content" active={selected.indexOf(item.id) !== -1}>
                <Menu text vertical>
                  <div className="menu-content">
                    {
                      item.option.map(option => (
                        <Menu.Item
                          key={option.id}
                          name={option.name}
                          as={NavLink}
                          to={option.path}
                        />
                      ))
                    }
                  </div>
                </Menu>
              </Accordion.Content>
            </div>
          ))
        }
      </Accordion>
      <style jsx="true">{styles}</style>
    </div>
  );
}
