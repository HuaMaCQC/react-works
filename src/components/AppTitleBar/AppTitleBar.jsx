import React from 'react';
import styles from './AppTitleBar.styledJsx.scss';

export default function AppTitleBar(props) {
  const { title } = props;

  return (
    <div className="app-title-bar">
      <div data-testid="title" className="bar-header">{title}</div>
      <style jsx="true">{styles}</style>
    </div>
  );
}
