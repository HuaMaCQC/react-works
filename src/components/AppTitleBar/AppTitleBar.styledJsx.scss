@import "../../App.scss";

.app-title-bar {
  width : 100%;
  height: 100%;
  padding: 10px;

  .bar-header {
    color: $color-primary-4;
    font-size: 24px;
    vertical-align: middle;
    display: inline-block;
  }
}
