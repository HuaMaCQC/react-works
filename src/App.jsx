import React, { useState } from 'react';
import { HashRouter, Route } from 'react-router-dom';
import AppHeader from './components/AppHeader';
import styles from './App.styledJsx.scss';
import AppMenu from './components/AppMenu';
import AppFooter from './components/AppFooter';
import Home from './components/Home/Home';
import MemberList from './components/member/memberList/MemberList';
import MemberEdit from './components/member/memberEdit/MemberEdit';
import AppTitleBar from './components/AppTitleBar/AppTitleBar';
import './App.scss';

export default function App() {
  const menu = [
    {
      id: 1,
      name: '會員管理',
      option: [
        {
          id: 1,
          name: '會員列表',
          path: '/member-table',
        },
        {
          id: 2,
          name: '會員編輯',
          path: '/member-edit',
        },
        {
          id: 3,
          name: '訊息',
          path: '/message',
        },
      ],
    },
    {
      id: 2,
      name: '站內信',
      option: [
        {
          id: 4,
          name: '信件',
          path: '/letter',
        },
      ],
    },
  ];
  const user = { id: 1, name: 'HuaMa' };
  const routeComponent = [
    { id: 1, component: MemberList, path: '/member-table' },
    { id: 2, component: MemberEdit, path: '/member-edit' },
  ];
  const [menuShow, setMenuShow] = useState(true);
  const [title, setTitle] = useState();

  return (
    <div className="app">
      <div className="header">
        <AppHeader switchMenu={() => setMenuShow(!menuShow)} user={user} />
      </div>
      <div className="main">
        <HashRouter>
          <div className={`${menuShow ? '' : 'hide'} menu`} data-testid="menu">
            <AppMenu menu={menu} setTitle={setTitle} />
          </div>
          <div className={`${menuShow ? '' : 'menu-hide'} article`}>
            <div className="box">
              <div className="explore-bar">
                <AppTitleBar title={title} />
              </div>
              <div className="component">
                <Route path="/" component={Home} exact />
                {routeComponent.map(route => (
                  <Route
                    key={route.id}
                    path={route.path}
                    component={route.component}
                  />
                ))}
              </div>
            </div>
          </div>
        </HashRouter>
      </div>
      <div className="footer bg-secondary-1">
        <AppFooter />
      </div>
      <style jsx="true">{styles}</style>
    </div>
  );
}
