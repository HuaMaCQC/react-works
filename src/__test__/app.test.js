/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import {
  cleanup,
  render,
} from '@testing-library/react';
import 'jest-dom/extend-expect';
import App from '../App';
import AppHeader from '../components/AppHeader';

import AppMenu from '../components/AppMenu';
import AppTitleBar from '../components/AppTitleBar/AppTitleBar';

// 處理錯誤資訊
const originalError = console.error;
beforeAll(() => {
  console.error = jest.fn();
});

afterAll(() => {
  console.error = originalError;
});

jest.mock('../components/AppHeader');
jest.mock('../components/AppMenu');
jest.mock('../components/AppTitleBar/AppTitleBar');

let headerPropsData = {};
const headerMockElement = (data) => {
  headerPropsData = data;
  return (<div />);
};
AppHeader.mockImplementation(headerMockElement);

let menuPropsData = {};
const menuMockElement = (data) => {
  menuPropsData = data;
  return (<div />);
};
AppMenu.mockImplementation(menuMockElement);

let titlePropsData = {};
const titleMockElement = (data) => {
  titlePropsData = data;
  return (<div />);
};
AppTitleBar.mockImplementation(titleMockElement);

afterEach(() => {
  cleanup();
  jest.clearAllMocks();
});

it('Snapshot Testing', () => {
  const { container } = render(<App />);

  expect(container).toMatchSnapshot();
});

it('test Menu ', () => {
  const { getByTestId } = render(<App />);

  expect(getByTestId('menu')).toBeTruthy();
  expect(getByTestId('menu')).not.toHaveClass('hide');

  headerPropsData.switchMenu();

  expect(getByTestId('menu')).toHaveClass('hide');
});

it('test title', () => {
  render(<App />);
  menuPropsData.setTitle('test');
  expect(titlePropsData.title).toBe('test');
});
