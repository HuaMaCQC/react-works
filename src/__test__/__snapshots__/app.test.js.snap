// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing 1`] = `
<div>
  <div
    class="app"
  >
    <div
      class="header"
    >
      <div />
    </div>
    <div
      class="main"
    >
      <div
        class=" menu"
        data-testid="menu"
      >
        <div />
      </div>
      <div
        class=" article"
      >
        <div
          class="box"
        >
          <div
            class="explore-bar"
          >
            <div />
          </div>
          <div
            class="component"
          >
            <div>
              welcome!!~~~~~~~~~~~~~~~~~~
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="footer bg-secondary-1"
    >
      <div />
    </div>
    <style
      jsx="true"
    >
      App.styledJsx.scss
    </style>
  </div>
</div>
`;
